src/Uno.UI/UI/Xaml/UIElement.Layout.netstd.cs

```diff
@@ -4,98 +4,204 @@
 using System.Diagnostics;
 using Windows.UI.Xaml.Controls.Primitives;
 using System.Runtime.CompilerServices;
+using Uno.UI;
 
 namespace Windows.UI.Xaml
 {
 	public partial class UIElement : DependencyObject
 	{
 		private Size _size;
 
-		private bool _isMeasureValid = false;
-		private bool _isArrangeValid = false;
-
 		public Size DesiredSize => Visibility == Visibility.Collapsed ? new Size(0, 0) : ((IUIElement)this).DesiredSize;
 
-		internal bool IsMeasureDirty => !_isMeasureValid;
-		internal bool IsArrangeDirty => !_isArrangeValid;
-
 		/// <summary>
-		/// When set, measure and invalidate requests will not be propagated further up the visual tree, ie they won't trigger a relayout.
+		/// When set, measure and invalidate requests will not be propagated further up the visual tree, ie they won't trigger a re-layout.
 		/// Used where repeated unnecessary measure/arrange passes would be unacceptable for performance (eg scrolling in a list).
 		/// </summary>
 		internal bool ShouldInterceptInvalidate { get; set; }
 
+		[Flags]
+		private protected enum LayoutFlag : byte
```

Comment:
I understand that those flags are named `LayoutFlag` since they are in the "Layout related partial class", but should we merge all "core flags that are going to be present on all platforms" in a single enum to reduce memory footprint ?

Reply:
They are doing it as you said in WinUI...  but my goal was not to rewrite all boolean flags in this PR.  I agree they can/should be merged.

Reply:
Indeed it could be in another PR, but then I would not name the enum `LayoutFlag`, I would choose a more generic name like `StateFlag` (even if I don't like that name neither)